### PR TITLE
Issue #3190: Allow contact form permissions to be set on the contact …

### PR DESCRIPTION
**Pull request by opi**
…settings page.

### CODE REVIEW

File: core/modules/contact/contact.admin.inc [hunk 1]
```diff
@@ -53,9 +62,95 @@ function contact_settings_form($form, &$form_state) {
...
+    '#value' => $roles,
+  );
+
+  // Retrieve contact related permissions.
+  $permissions = contact_permission();
+
+  // Render role/permission overview:
+  $options = array();
+  $status = array();
+  foreach ($permissions as $perm => $perm_item) {
```
Comment (klonos) [inline core/modules/contact/contact.admin.inc:113]: I would use full names for variables. Makes them less "cryptic" and avoids ambiguities when these variables are being used later in the code (perm = "permission" or "permanent").

Comment (opi) [reply core/modules/contact/contact.admin.inc:113]: Agreed.
However, this is a copy/paste of `node_type_form_permissions()`. I'm in favor to open a PR for both then.

Comment (opi) [reply core/modules/contact/contact.admin.inc:113]: oh, 4 of them !
```bash
$ git grep "foreach (\$permissions as \$perm => \$perm_item)"
core/modules/node/node.types.inc:  foreach ($permissions as $perm => $perm_item) {
core/modules/taxonomy/taxonomy.admin.inc:  foreach ($permissions as $perm => $perm_item) {
core/modules/user/user.admin.inc:      foreach ($permissions as $perm => $perm_item) {
core/modules/user/views/views_handler_filter_user_permissions.inc:        foreach ($permissions as $perm => $perm_item) {
```

**Comment by herbdool** [comment]
This looks good! I tested and works well.

I agree about making the variable names a separate issue. It's less of an issue when it's used in foreach loop. I've seen it used a lot like that

**Comment by quicksketch** [comment]
Thanks @herbdool and @klonos for the feedback. Agreed on the reasons stated here. This looks good to me and works great in the sandbox! Merged into 1.x for 1.11.0.

**Pull request merged**